Tasks/GooglePlayStatusUpdate/google-play-status-update.ts

```diff
@@ -0,0 +1,88 @@
+import * as path from 'path';
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as googleutil from './googleutil';
+
+async function run() {
+    try {
+        tl.setResourcePath(path.join(__dirname, 'task.json'));
+
+        const authType: string = tl.getInput('authType', true);
+        let key: googleutil.ClientKey = {};
+        if (authType === 'JsonFile') {
+            const serviceAccountKeyFile: string = tl.getPathInput('serviceAccountKey', false);
+            if (!serviceAccountKeyFile) {
+                throw new Error(tl.loc('JsonKeyFileNotFound'));
+            }
+            const stats: tl.FsStats = tl.stats(serviceAccountKeyFile);
+            if (stats && stats.isFile()) {
+                key = require(serviceAccountKeyFile);
+            } else {
+                console.error(tl.loc('InvalidAuthFile'));
+                throw new Error(tl.loc('InvalidAuthFilewithName', serviceAccountKeyFile));
+            }
+        } else if (authType === 'ServiceEndpoint') {
+            const serviceEndpoint: tl.EndpointAuthorization = tl.getEndpointAuthorization(tl.getInput('serviceEndpoint', true), true);
+            if (!serviceEndpoint) {
+                throw new Error(tl.loc('EndpointNotFound'));
+            }
+            key.client_email = serviceEndpoint.parameters['username'];
+            key.private_key = serviceEndpoint.parameters['password'].replace(/\\n/g, '\n');
+        }
+
+        const packageName: string = tl.getPathInput('packageName', true);
+        const status: string = tl.getInput('status', true);
+        const userFrac: string = tl.getInput('userFraction', false);
+        let userFraction: number = Number.NaN;
```

Comment:
NIT: I think it's confusing to keep two very similarly named variables together. To make it clearer, `userFrac` could be renamed to `userFractionInput`.

Reply:
Thanks, I have renamed it to `userFractionInput`.